modules/seqexec/server/src/main/scala/seqexec/server/ghost/GHOST.scala

```diff
@@ -0,0 +1,151 @@
+import seqexec.server.ghost.GHOSTController._
+import squants.time.{Seconds, Time}
+
+final case class GHOST[F[_]: Sync](controller: GHOSTController[F])
+    extends InstrumentSystem[F]
+    with GDSInstrument {
+  override val gdsClient: GDSClient = controller.gdsClient
+
+  override val resource: Resource = Instrument.GHOST
+
+  override val sfName: String = "GHOST"
+
+  override val contributorName: String = "ghost"
+
+  override val observeControl: InstrumentSystem.ObserveControl =
+    InstrumentSystem.Uncontrollable
+
+  override def observe(
+      config: Config): SeqObserveF[F, ImageFileId, ObserveCommand.Result] =
+    Reader { fileId =>
+      controller
+        .observe(fileId)
+        .map(_ => ObserveCommand.Success: ObserveCommand.Result)
+    }
+
+  override def configure(config: Config): SeqActionF[F, ConfigResult[F]] =
+    GHOST
+      .fromSequenceConfig[F](config)
+      .flatMap(controller.applyConfig)
+      .map(_ => ConfigResult[F](this))
+
+  override def notifyObserveEnd: SeqActionF[F, Unit] = controller.endObserve
+
+  override def notifyObserveStart: SeqActionF[F, Unit] = SeqActionF.void
+
+  override def calcObserveTime(config: Config): Time = Seconds(360)
+}
+
+object GHOST {
+  val INSTRUMENT_NAME_PROP: String = "GHOST"
+  val name: String                 = INSTRUMENT_NAME_PROP
+
+  val sfName: String = "GHOST"
+
+  // TODO: I am sure there is a cleaner way to write this without using for, but I'm not sure right
+  // TODO: now what it is.
+//  private def hmsAngle(config: Config, name: String): Either[ExtractFailure, Option[HourAngle]] =
+//    for {
+//      angOptStr <- config.extract(INSTRUMENT_KEY / name).as[Option[String]]
+//      angStr    <- angOptStr
+//      ang       <- HourAngle.fromStringHMS.getOption(angStr)
+//    } yield ang
+//  private def hmsAngle(config: Config, name: String): Either[ExtractFailure, Option[HourAngle]] =
+//    for {
+//      a <- config.extract(INSTRUMENT_KEY / name).as[String]
+//          .map(HourAngle.fromStringHMS.getOption)
+//    } yield a
+//
+//  private def dmsAngle(config: Config, name: String): Either[ExtractFailure, Option[Angle]] =
+//    for {
+//      a <- config.extract(INSTRUMENT_KEY / name).as[String]
+//        .map(Angle.fromStringSignedDMS.getOption)
+//    } yield a
+//
+//
+//  private def targetName(config: Config, name: String): Either[ExtractFailure, Option[String]] =
+//    config.extract(INSTRUMENT_KEY / name).as[Option[String]]
+
+  def fromSequenceConfig[F[_]: Sync](
+      config: Config): SeqActionF[F, GHOSTConfig] =
+    EitherT(
+      Sync[F].delay(
+        (for {
+          baseRAHMS          <- config
+                                 .extract(INSTRUMENT_KEY / Ghost.BaseRAHMS)
+                                 .as[Option[HourAngle]]
+          baseDecDMS         <- config
+                                 .extract(INSTRUMENT_KEY / Ghost.BaseDecDMS)
+                                 .as[Option[Angle]]
```

Comment:
I suppose you could add an `extractAs[A](...)` and make these 1 liners.

Reply:
Right, that's a good idea, though we won't need these with the new model

Reply:
Good point.  I only mention it because of the scala check method length test.

Reply:
yes, we keep hitting that limit in these methods. I'd rather remove the 50 lines limit check in scalastyle

Reply:
Yeah the boilerplate could be reduced here, and you could use `mapN` instead of a for-comprehension if you wanted to (not a big deal), I would rather keep the 50-line limit and disable it for cases where you have a good reason to have a long method. It's a good stylistic check.

Reply:
Actually, it was very easy to add and looks a lot more compact. thanks for the idea